morpheus/modules/filter_control_message.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+import mrc
+from mrc.core import operators as ops
+
+from morpheus.messages import MessageControl
+from morpheus.utils.module_ids import FILTER_CONTROL_MESSAGE
+from morpheus.utils.module_ids import MODULE_NAMESPACE
+from morpheus.utils.module_utils import get_module_config
+from morpheus.utils.module_utils import register_module
+
+logger = logging.getLogger(__name__)
+
+
+@register_module(FILTER_CONTROL_MESSAGE, MODULE_NAMESPACE)
+def filter_control_message(builder: mrc.Builder):
+    """
+    When the requirements are met, this module gently discards the control messages.
+
+    Parameters
+    ----------
+    builder : mrc.Builder
+        mrc Builder object.
+    """
+
+    config = get_module_config(FILTER_CONTROL_MESSAGE, builder)
+
+    filter = config.get("data_type", None)
+    enable_task_check = config.get("enable_task_check", False)
+
+    def on_data(control_message: MessageControl):
+        data_type = control_message.get_metadata("data_type")
+
+        if enable_task_check:
+            tasks = control_message.config().get("tasks")
```

Comment:
I'm going to get ride of the 'config()' method in the near future to lock things down a bit more. Instead of actually grabbing tasks, this can be checked with `control_message.has_task('inference')` or `control_message.has_task('training')`